compiler/src/iree/compiler/Codegen/Common/TransformDialectStrategies.cpp

```diff
@@ -73,6 +74,15 @@ int64_t mlir::iree_compiler::nextMultipleOf(int64_t val, int64_t multiple) {
   return ((val + multiple - 1) / multiple) * multiple;
 }
 
+FailureOr<int64_t> mlir::iree_compiler::maxDivisorOfValueBelowLimit(
+    int64_t value, int64_t limit) {
+  assert(limit <= 1024 && "limit must be smaller than 1024");
```

Comment:
Why an arbitrary assert here? Seems like a foot gun? This function seems to have a reasonable ability to fail. Why not do that?